syntax = "proto3";
package WY;

import "msg.proto";

// 信道的打开与关闭
message openChannelRequest
{
    // 对于请求我们都是需要给予一个响应的，在高并发下，会出现很多消息
    // 因此我们需要为每一个请求都加上请求id，来进行区分，方便给予响应时可以表明响应是哪一条消息
    string rid = 1;  // 请求id          
    string cid = 2;  // 信道id
}
message closeChannelRequest
{
    string rid = 1;  
    string cid = 2;
}

// 交换机的声明和删除
message declareExchangeRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    ExchangeType exchange_type = 4;
    bool durable = 5;
    bool auto_delete = 6;
    map<string, string> args = 7;
}
message deleteExchangeRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
}

// 队列的声明和删除
message declareQueueRequest
{
    string rid = 1;
    string cid = 2;
    string queue_name = 3;
    bool exclusive = 4;    // 是否独占标志
    bool durable = 5;
    bool auto_delete = 6;
    map<string, string> args = 7;
} 
message deleteQueueRequest
{
    string rid = 1;
    string cid = 2;
    string queue_name = 3;
}

// 队列的绑定和解绑
message queueBindRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    string queue_name = 4;
    string binding_key = 5;
}
message queueUnBindRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    string queue_name = 4;
}

// 消息的发布
message basicPublishRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    string body = 4;
    BasicProperties properties = 5;
}

// 消息的确认
message basicAckRequest
{
    string rid = 1;
    string cid = 2;
    string queue_name = 3;
    string message_id = 4;
}

// 队列的订阅
message basicConsumeRequest
{
    string rid = 1;
    string cid = 2;
    string consumer_tag = 3;
    string queue_name = 4;
    bool auto_ack = 5;
}

// 订阅的取消
message basicCancelRequest
{
    string rid = 1;
    string cid = 2;
    string consumer_tag = 3;
    string queue_name = 4;
}

// 消息的推送
message basicConsumeResponse
{
    string cid = 1;
    string consumer_tag = 2;
    string body = 3;
    BasicProperties properties = 4;
}

// 通用的响应
message basicCommonResponse
{
    string rid = 1;
    string cid = 2;
    bool ok = 3;
}